### DISCUSSION TITLE
Queues are not tenant aware

### DISCUSSION THREAD

**Discussion by ju5t** [spatie/laravel-multitenancy#409] [Q&A]
I have a weird situation with our application. We have a `Mailable` that implements `ShouldQueue`. We send the e-mail through a specific tenant as follows below:

```PHP
$tenant->execute(fn () => Mail::to($email)->send(new QuoteMail($data)));
```

`queues_are_tenant_aware_by_default` is set to true in our installation. But still, when the email goes out, it has no tenant. Even when I explicitly make it inherit `TenantAware`, it doesn't work.

We had something similar in the past. Scheduled jobs that do not require a tenant would not work even when they inherited `NotTenantAware`. I had to add them to the `not_tenant_aware_jobs` array in the config. That worked.

When I add the email to the `tenant_aware_jobs`, it gets the correct tenant, but that will get tedious over time. I'll do some debugging when I'm at work next week, but perhaps it's something simple I've missed? We're on 2.3.5, but the release notes didn't mention anything that might cause this.

**Comment by r9software** [comment]
Did you found anything?
